### PR TITLE
Support select operator

### PR DESCRIPTION
**Pull request by dschuff**
Select is a conditional operator which evaluates a condition and 2 input
operands, and returns the first if the condition is nonzero, and the
second otherwise.

### CODE REVIEW

File: ml-proto/host/lexer.mll [hunk 1]
```diff
@@ -231,6 +231,8 @@ rule token = parse
     { CONVERT (floatop t Float32Op.ConvertSInt64 Float64Op.ConvertSInt64) }
   | (fxx as t)".convert_u/i64"
     { CONVERT (floatop t Float32Op.ConvertUInt64 Float64Op.ConvertUInt64) }
+  | (ixx as t)".select" { SELECT ( intop t Int32Op.Select Int64Op.Select) }
```
Comment (rossberg-chromium) [inline ml-proto/host/lexer.mll]: Nit: can you move these to a more consistent place (i.e., not in the middle of the conversion operators)?

File: ml-proto/spec/check.ml [hunk 2]
```diff
@@ -211,6 +212,12 @@ let rec check_expr c et e =
     check_expr c (Some t1) e1;
     check_type (Some t) et e.at
 
+  | Select (selectop, e1, e2, e3) ->
+    let t = type_selectop selectop in
+    check_expr c (Some Int32Type) e1;
+    check_expr c (Some t) e2;
+    check_expr c (Some t) e3;
```
Comment (rossberg-chromium) [inline ml-proto/spec/check.ml]: Nit: stray semicolon

File: ml-proto/spec/sugar.ml [hunk 3]
```diff
@@ -85,6 +85,9 @@ let compare (relop, e1, e2) =
 let convert (cvt, e) =
   Convert (cvt, e)
 
+let select (selectop, cond, e1, e2) =
+  Select(selectop, cond, e1, e2)
```
Comment (rossberg-chromium) [inline ml-proto/spec/sugar.ml]: Nit: spacing

**Comment by rossberg-chromium** [comment]
LGTM modulo nits

File: ml-proto/host/lexer.mll (hunk 1 above)
Comment (dschuff) [inline ml-proto/host/lexer.mll]: Done.

File: ml-proto/spec/check.ml (hunk 2 above)
Comment (dschuff) [inline ml-proto/spec/check.ml]: Done.

File: ml-proto/spec/sugar.ml (hunk 3 above)
Comment (dschuff) [inline ml-proto/spec/sugar.ml]: Done.

**Comment by dschuff** [comment]
Thanks!

**Pull request merged**